File: abc/abc156/abc156_c.cs
// https://atcoder.jp/contests/abc156/submissions/10293085

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        static void Main()
        {
            var N = int.Parse(Console.ReadLine());
            var X = Console.ReadLine().Split().Select(long.Parse).ToArray();

            var p = (long)Math.Round(X.Average());

            Console.WriteLine($"{X.Select(x => (x - p) * (x - p)).Sum()}");
        }
    }
}

File: abc/abc156/abc156_c.fs
// https://atcoder.jp/contests/abc156/submissions/11878988

let N = stdin.ReadLine() |> int
let X = stdin.ReadLine().Split() |> Array.map int64

let p =
    X
    |> Array.averageBy float
    |> round
    |> int64

X
|> Array.sumBy (fun x -> (x - p) * (x - p))
|> stdout.WriteLine
